store/tx_manager.go

```diff
@@ -177,11 +195,60 @@ func (txm *TxManager) handleUnconfirmed(
 
 func (txm *TxManager) bumpGas(txat *models.TxAttempt, blkNum uint64) error {
 	tx := &models.Tx{}
-	if err := txm.ORM.One("ID", txat.TxID, tx); err != nil {
+	if err := txm.orm.One("ID", txat.TxID, tx); err != nil {
 		return err
 	}
-	gasPrice := new(big.Int).Add(txat.GasPrice, &txm.Config.EthGasBumpWei)
+	gasPrice := new(big.Int).Add(txat.GasPrice, &txm.config.EthGasBumpWei)
 	txat, err := txm.createAttempt(tx, gasPrice, blkNum)
 	logger.Infow(fmt.Sprintf("Bumping gas to %v for transaction %v", gasPrice, txat.Hash.String()), "txat", txat)
 	return err
 }
+
+// GetActiveAccount returns a copy of the TxManager's active nonce managed
+// account.
+func (txm *TxManager) GetActiveAccount() *ActiveAccount {
+	if txm.activeAccount == nil {
+		return nil
+	}
+	dup := *txm.activeAccount
```

Comment:
Code Climate is complaining: " assignment copies lock value to dup: store.ActiveAccount contains sync.Mutex"

That is the functionality we want though, right?

Reply:
The issue is that `dup.GetAndIncrementNonce()` would lock the original `txm.activeAccount` as well. We never call `dup.GetAndIncrementNonce()` though, we only use dup for `accounts.Account`, and `nonce`.

To be on the safe side, I would listen to the code climate recommendation and just explicitly copy the two fields `nonce`, and `Account`, omitting `mutex`.

Reply:
Fixed by explicitly copying the struct members